packages/components/src/components/tooltip/component.tsx

```diff
@@ -94,24 +97,30 @@ export class KolTooltip implements API {
 	private addListeners = (el: Element): void => {
 		el.addEventListener('mouseover', this.incrementOverFocusCount);
 		el.addEventListener('focus', this.incrementOverFocusCount);
+		el.addEventListener('focusin', this.incrementOverFocusCount);
 		el.addEventListener('mouseout', this.decrementOverFocusCount);
 		el.addEventListener('blur', this.decrementOverFocusCount);
+		el.addEventListener('focusout', this.decrementOverFocusCount);
 	};
 
 	private removeListeners = (el: Element): void => {
 		el.removeEventListener('mouseover', this.incrementOverFocusCount);
 		el.removeEventListener('focus', this.incrementOverFocusCount);
+		el.removeEventListener('focusin', this.incrementOverFocusCount);
 		el.removeEventListener('mouseout', this.decrementOverFocusCount);
 		el.removeEventListener('blur', this.decrementOverFocusCount);
+		el.addEventListener('focusout', this.decrementOverFocusCount);
 	};
 
-	private resyncListeners = (el: Element): void => {
-		this.removeListeners(el);
-		this.addListeners(el);
-	};
-
-	private catchHostElement = (el: HTMLElement | null): void => {
-		this.previousSibling = el?.previousElementSibling as HTMLElement | null;
+	private resyncListeners = (last?: Element | null, next?: Element | null): void => {
+		console.log('resyncListeners', last, next);
+		if (last) {
+			this.removeListeners(last);
+		}
+		if (next) {
+			last = next;
```

Comment:
Wozu diese Zeile?

Reply:
Das brauchen wir, damit die Referenz auf den `next` Container im nächsten Durchlauf der `last` ist. Muss dokumentiert werden.

Reply:
Vielleicht lese ich es falsch, aber wird hier nicht nur die lokale Referenz `last` umgeschrieben, die danach nicht mehr verwendet wird?

Reply:
Danke, dass ist überhaupt nicht gut zu lesen. Die beiden Parameter `last` und `next` sind Referenzen auf die `this.*`-Properties. Deshalb setzt `last = next` gleichzeitig `this.* = next`.

Reply:
I hope, it is better now.